go-controller/pkg/ovn/policy.go

```diff
@@ -635,14 +643,14 @@ func (oc *Controller) handleLocalPodSelectorAddFunc(
 		return
 	}
 
-	np.Lock()
-	defer np.Unlock()
+	np.RLock()
+	defer np.RUnlock()
 
 	if np.deleted {
 		return
 	}
 
-	if _, ok := np.localPods[logicalPort]; ok {
+	if _, ok := np.localPods.Load(logicalPort); ok {
```

Comment:
Can't we move this step up and out of the Lock `np.RLock()` since by design sync.Maps are threadsafe so we don't need to hold the lock on `np` while we operate on the map

Reply:
Oh, definitely, good point.

Reply:
But we still need the RLock() to read `.deleted`